### PR TITLE
llvm36: Patch to fix --with-asan

### LINKED ISSUE #756: llvm36 fail to build in 10.10 if add --with-asan
Latest Command Line Tools is installed.

Output of `brew install llvm36 --with-lldb --with-lld --with-all-targets --with-asan`:

<https://gist.github.com/anonymous/b241e76a120ecccd76c3>

    make[3]: *** [compiler-rt/.makeall] Error 2
    make[2]: *** [all] Error 1
    make[1]: *** [clang/.makeall] Error 2
    make: *** [all] Error 1

It fail to build because of option "--with-asan", if i use `brew install llvm36 --with-lldb --with-lld --with-all-targets`, it will success to build.

So where the problem is?

### PR DESCRIPTION
**Pull request by flaub**
Fixes #756.

### CODE REVIEW

File: llvm36.rb [hunk 1]
```diff
@@ -143,6 +143,7 @@ def install
     (buildpath/"tools/lld").install resource("lld") if build.with? "lld"
     (buildpath/"tools/lldb").install resource("lldb") if build.with? "lldb"
     (buildpath/"projects/compiler-rt").install resource("compiler-rt") if build.with? "asan"
+    inreplace "tools/clang/runtime/compiler-rt/Makefile", /CC=\"\$\(ToolDir\)\/clang\" \\/, "\\" if build.with? "asan"
```
Comment (flaub) [inline llvm36.rb]: I don't know if this is the best way to accomplish this. I want to patch the file but wasn't clear about how to use an embedded patch.

**Comment by flaub** [comment]
This can likely be applied to llvm34 and llvm35 as well.

**Comment by DomT4** [comment]
We've reverted various workarounds for `asan` before now, so will let other @Homebrew/owners have some time to review here.

File: llvm36.rb (hunk 1 above)
Comment (DomT4) [inline llvm36.rb]: Can we report the need for doing this upstream? I'd like to keep track on it long-term. Perhaps the LLVM team can flag up whether this is something we're causing or something that's a core LLVM issue.

You can also block the two "asan" calls together, like this:

```ruby
if build.with? "asan"
  (buildpath/"projects/compiler-rt").install resource("compiler-rt")
  # https://bugreport.submitted.com - Fix asan compile
  inreplace "tools/clang/runtime/compiler-rt/Makefile", /CC=\"\$\(ToolDir\)\/clang\" \\/, "\\"
end
```

**Comment by apjanke** [comment]
Ping on this? It's been a year, so I'm inclined to close this unless there's been upstream bug report activity.

**Pull request closed without merging**